doc/installation.rst

```diff
@@ -0,0 +1,267 @@
+Installation Guide
+==================
+
+This guide covers the installation and setup of the FoamAdapter Python module, which provides a Python interface to OpenFOAM solvers and utilities through the pybFoam backend.
+
+Prerequisites
+-------------
+
+System Requirements
+~~~~~~~~~~~~~~~~~~~
+
+Before installing FoamAdapter Python module, ensure your system meets the following requirements:
+
+* **Operating System**: Linux (Ubuntu 20.04+, CentOS 8+, or similar)
+* **Python**: 3.9 or higher
+* **OpenFOAM**: Version 2406 or later
+* **CMake**: 3.22 or higher
+* **Compiler**: GCC 12+ or Clang 18+
+* **CUDA**: 12.1+ (optional, for GPU acceleration)
+
+Python Dependencies
+~~~~~~~~~~~~~~~~~~~
+
+The FoamAdapter Python module requires the following Python packages:
+
+.. code-block:: text
+
+    pybFoam>=1.0.0
+    pydantic>=2.0.0
+    typer>=0.9.0
+    matplotlib>=3.6.0
+    networkx>=2.8.0
+
+Installation Methods
+--------------------
+
+Development Installation (Recommended)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For development or if you want the latest features:
+
+1. **Clone the repository**:
+
+   .. code-block:: bash
+
+       git clone https://github.com/exasim-project/FoamAdapter.git
+       cd FoamAdapter
+
+2. **Set up Python environment**:
+
+   .. code-block:: bash
+
+       # Using conda (recommended)
+       conda create -n foamadapter python=3.11
+       conda activate foamadapter
+       
+       # Or using venv
+       python -m venv foamadapter-env
+       source foamadapter-env/bin/activate
+
+3. **Install dependencies**:
+
+   .. code-block:: bash
+
+       pip install -r requirements.txt
+
+4. **Build the C++ backend**:
+
+   .. code-block:: bash
+
+       cmake --preset production
+       cmake --build --preset production
+
+5. **Install the Python module**:
+
+   .. code-block:: bash
+
+       pip install -e ./src
```

Comment:
Switch 4 and 5. I think if we are going to be more pythonic we should drive cmake from pip. And only mention the cmake step as optional.